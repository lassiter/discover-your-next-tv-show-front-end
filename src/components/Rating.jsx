import React, { Component } from 'react'
import styled from 'styled-components'
import star from '../img/star.png'

const Wrapper = styled.div`
  display: flex;
`

const Rate = styled.h1`
  color: white;
  display: flex;
  font-size: 50px;
  margin-block-start: 0em;
  margin-block-end: 00em;
  margin-inline-start: 0px;
  margin-inline-end: 0px;
  font-weight: bold;
`
const Star = styled.img`
  margin-top: 10px;
  width: 15px;
  height: 15px;
`
const InternalWrapper = styled.div`
  display: flex;
  flex-flow: column;
  & > small {
    color: white;
    margin: auto;
  }
`

export default class Rating extends Component {
  render() {
    return (
      <Wrapper>
        <Rate>{this.props.rating}</Rate>
        <InternalWrapper>
          <Star src={star}/>
          <small>/10</small>
        </InternalWrapper>
      </Wrapper>
    )
  }
}
