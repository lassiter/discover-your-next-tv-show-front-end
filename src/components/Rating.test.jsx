import React from 'react';
import Rating from './Rating';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const voteAverage = 8.2
  const voteCount = 100
  const tree = renderer
    .create(<Rating rating={voteAverage} votes={voteCount}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});