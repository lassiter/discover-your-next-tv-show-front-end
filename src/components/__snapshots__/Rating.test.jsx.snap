// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa kZAgjJ"
>
  <h1
    className="sc-bwzfXH bSBkPi"
  >
    8.2
  </h1>
  <div
    className="sc-bxivhb eugCME"
  >
    <img
      className="sc-htpNat ctJLlT"
      src="star.png"
    />
    <small>
      /10
    </small>
  </div>
</div>
`;
